File: src/main/java/tr/com/ias/entity/iasHtmlButton.java
package tr.com.ias.entity;

public class iasHtmlButton extends iasHtmlComponent {

	private String buttonColor;

	public iasHtmlButton() {
		super();
		componentType = ComponentType.BUTTON;
	}

	public iasHtmlButton(String buttonColor) {
		this();
		
		this.buttonColor = buttonColor;
	}

	public String getButtonColor() {
		return buttonColor;
	}

	public void setButtonColor(String buttonColor) {
		this.buttonColor = buttonColor;
	}
}

File: front_end/dynamic-component/src/app/ias-html-button.ts
import { iasHtmlComponent } from './ias-html-component';
import { ComponentType } from './component-type';

export class iasHtmlButton extends iasHtmlComponent{
	buttonColor: string;
	  constructor(private e: JSON) {
	  	super();
		  this.componentType = ComponentType.BUTTON;
		  this.x = e['x'];
		  this.y = e['y'];
		  this.z = e['z'];
		  this.caption = e['caption'];
		  this.buttonColor = e['buttonColor'];
	   }

}